### ISSUE TITLE
fix: TypeDI update broke `useContainer`

### ISSUE THREAD

**Issue by EirikFA** [typestack/class-validator#928]
### Description
A recent update to TypeDI (> 0.8.0) introduced a change that requires all classes to be decorated with `@Service()`. This means that calling `useContainer` from `class-validator` will not work anymore. Services could now be registered in the container when it is first recorded (https://github.com/typestack/typedi/issues/173#issuecomment-783627227).

**Minimal code-snippet showcasing the problem**
 

```ts
import { useContainer } from "class-validator";
import { Container } from "typedi";

// Throws ServiceNotFoundError for MaybeConstructable<Validator> when used
useContainer(Container);
```

### Expected behavior
All classes that are required should be registered when using `useContainer`.

### Actual behavior
No classes are registered, thus dependency injection with `class-validator` does not work.

**Comment by NoNameProvided** [comment]
Yes this is expected. The internal container needs to be updated to something like this: https://github.com/typeorm/typeorm-typedi-extensions/blob/master/src/container-provider.class.ts. However since this would put a hard dependency on TypeDI it should be written in a way where it checks for each function before calling them and only calls them if they exists. 

Just as a side note, I don't find the `useContainer` method fitting for class-validator. I would rather introduce an API to specifing external input for a validation. (Users can still pass in their container if they want to.)

**Comment by EirikFA** [comment]
Makes sense. I'm just thinking out loud here, but it might be possible to accept a function to `useContainer` that specifies how types shall be registered. Something like this
```ts
useContainer(
  Container,
  constructable => Container.set({ id: constructable, type: constructable as Constructable<T> })
);
```
Again, it might be completely irrelevant but I'm just throwing it out in case it might be useful.

**Comment by vavra7** [comment]
Any update on this or workaround? I am using typeorm-typedi-extensions and can not get my typeorm repository into my custom validation because class-validator is now not possible to use with typedi container.

**Comment by jpabeem** [comment]
What helped in our case was adding the `fallback` and `fallbackOnErrors` properties:

```ts
useClassValidatorContainer(Container, {
    fallback: true,
    fallbackOnErrors: true
});
```
